File: Practica 1.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>


int* generar_arreglo(int n) {
    int* arreglo = (int*)malloc(n * sizeof(int));
    srand(time(NULL));
    for (int i = 0; i < n; i++) {
        arreglo[i] = rand() % 1000 + 1;
    }
    return arreglo;
}


void imprimir_arreglo(int* arreglo, int n) {
    printf("Contenido del arreglo:\n");
    for (int i = 0; i < n; i++) {
        printf("%d ", arreglo[i]);
    }
    printf("\n");
}


int busqueda_secuencial(int* arreglo, int n, int elemento) {
    for (int i = 0; i < n; i++) {
        if (arreglo[i] == elemento) {
            return i;
        }
    }
    return -1;
}


void ordenar_arreglo(int* arreglo, int n) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (arreglo[j] > arreglo[j + 1]) {
                int temp = arreglo[j];
                arreglo[j] = arreglo[j + 1];
                arreglo[j + 1] = temp;
            }
        }
    }
}


void calcular_tiempo_ejecucion(int* arreglo, int n) {
    clock_t inicio = clock();
    generar_arreglo(1000);
    imprimir_arreglo(arreglo, n);
    busqueda_secuencial(arreglo, n, arreglo[0]);
    ordenar_arreglo(arreglo, n);
    busqueda_secuencial(arreglo, n, arreglo[0]);
    clock_t fin = clock();
    double tiempo_total = (double)(fin - inicio) / CLOCKS_PER_SEC;
    printf("Tiempo de ejecución de las operaciones: %f segundos\n", tiempo_total);
}


void menu() {
    printf("1. Generar arreglo aleatorio\n");
    printf("2. Imprimir arreglo\n");
    printf("3. Busqueda secuencial\n");
    printf("4. Ordenar arreglo\n");
    printf("5. Calcular tiempo de ejecucion\n");
    printf("6. Salir\n");
}

int main() {
    int* arreglo = NULL;
    int n = 1000;
    int opcion;
    while (1) {
        menu();
        printf("Ingrese su opcion: ");
        scanf("%d", &opcion);

        switch (opcion) {
            case 1:
                arreglo = generar_arreglo(n);
                break;
            case 2:
                if (arreglo != NULL) {
                    imprimir_arreglo(arreglo, n);
                } else {
                    printf("Debe generar un arreglo primero\n");
                }
                break;
            case 3:
                if (arreglo != NULL) {
                    int elemento_buscar;
                    printf("Ingrese el elemento a buscar: ");
                    scanf("%d", &elemento_buscar);
                    int indice_encontrado = busqueda_secuencial(arreglo, n, elemento_buscar);
                    if (indice_encontrado != -1) {
                        printf("El elemento %d se encuentra en el indice %d\n", elemento_buscar, indice_encontrado);
                    } else {
                        printf("El elemento %d no se encuentra en el arreglo\n", elemento_buscar);
                    }
                } else {
                    printf("Debe generar un arreglo primero\n");
                }
                break;
            case 4:
                if (arreglo != NULL) {
                    ordenar_arreglo(arreglo, n);
                    printf("Arreglo ordenado:\n");
                    imprimir_arreglo(arreglo, n);
                } else {
                    printf("Debe generar un arreglo primero\n");
                }
                break;
            case 5:
                if (arreglo != NULL) {
                    calcular_tiempo_ejecucion(arreglo, n);
                } else {
                    printf("Debe generar un arreglo primero\n");
                }
                break;
            case 6:
                printf("¡Adios!\n");
                free(arreglo);
                return 0;
            default:
                printf("Opcion no válida. Por favor, seleccione una opción del menu.\n");
                break;
        }
    }
}

File: Practica 1.py
import random
import time

# Función para generar un arreglo de números enteros aleatorios
def generar_arreglo(n):
    return [random.randint(1, 1000) for _ in range(n)]

# Función para imprimir el contenido de un arreglo
def imprimir_arreglo(arreglo):
    print("Contenido del arreglo:")
    print(arreglo)

# Función de búsqueda secuencial
def busqueda_secuencial(arreglo, elemento):
    for i in range(len(arreglo)):
        if arreglo[i] == elemento:
            return i
    return -1

# Función de ordenamiento de arreglo
def ordenar_arreglo(arreglo):
    arreglo.sort()

# Función para calcular el tiempo de ejecución
def calcular_tiempo_ejecucion(arreglo):
    inicio = time.time()
    generar_arreglo(1000)
    imprimir_arreglo(arreglo)
    busqueda_secuencial(arreglo, arreglo[0])
    ordenar_arreglo(arreglo)
    busqueda_secuencial(arreglo, arreglo[0])
    fin = time.time()
    tiempo_total = fin - inicio
    print("Tiempo de ejecución de las operaciones: ", tiempo_total, "segundos")

# Menú principal
def menu():
    print("1. Generar arreglo aleatorio")
    print("2. Imprimir arreglo")
    print("3. Búsqueda secuencial")
    print("4. Ordenar arreglo")
    print("5. Calcular tiempo de ejecución")
    print("6. Salir")

arreglo = []
while True:
    menu()
    opcion = int(input("Ingrese su opción: "))
    
    if opcion == 1:
        arreglo = generar_arreglo(1000)
    elif opcion == 2:
        if arreglo:
            imprimir_arreglo(arreglo)
        else:
            print("Debe generar un arreglo primero")
    elif opcion == 3:
        if arreglo:
            elemento_buscar = int(input("Ingrese el elemento a buscar: "))
            indice_encontrado = busqueda_secuencial(arreglo, elemento_buscar)
            if indice_encontrado != -1:
                print(f"El elemento {elemento_buscar} se encuentra en el índice {indice_encontrado}")
            else:
                print(f"El elemento {elemento_buscar} no se encuentra en el arreglo")
        else:
            print("Debe generar un arreglo primero")
    elif opcion == 4:
        if arreglo:
            ordenar_arreglo(arreglo)
            print("Arreglo ordenado:")
            imprimir_arreglo(arreglo)
        else:
            print("Debe generar un arreglo primero")
    elif opcion == 5:
        if arreglo:
            calcular_tiempo_ejecucion(arreglo)
        else:
            print("Debe generar un arreglo primero")
    elif opcion == 6:
        print("¡Adiós!")
        break
    else:
        print("Opción no válida. Por favor, seleccione una opción del menú.")
